analysis.ml

```diff
@@ -1,34 +1,44 @@
 open Instr
 
-let successors (instrs : instruction_stream) pc =
+let pcs (instrs : instruction_stream) : pc array =
+  Array.mapi (fun pc _ -> pc) instrs
```

Comment:
Of course, I was thinking of the error messages.

Your one-liner is fine, I think it is actually the best you can do. Another one is `Array.init (Array.length instrs) (fun i -> i)`. Some extended standard libraries (I mention a couple ones below) have other functions, but I think `mapi` is appropriate here.